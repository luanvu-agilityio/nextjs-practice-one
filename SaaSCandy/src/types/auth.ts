import { User } from './user';
export type AuthState =
  | {
      success: boolean;
      user: User;
      credentials: {
        email: string;
        password: string;
      };
      error: undefined;
    }
  | {
      error: unknown;
      success: undefined;
      user: undefined;
      credentials: undefined;
    };
