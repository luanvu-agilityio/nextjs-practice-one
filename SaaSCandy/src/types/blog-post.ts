import { ReactNode } from 'react';

export interface HeadingContent {
  level: 1 | 2 | 3 | 4 | 5 | 6;
  text: string;
}

export interface ParagraphContent {
  text: string;
}

export interface ListContent {
  items: string[];
  ordered: boolean;
}

export interface QuoteContent {
  text: string;
  author?: string;
}

export interface ImageContent {
  src: string;
  alt: string;
  width?: number;
  height?: number;
  caption?: string;
}

export interface CodeContent {
  code: string;
  language?: string;
}

export type ContentBlockData =
  | { type: 'heading'; content: HeadingContent }
  | { type: 'paragraph'; content: ParagraphContent }
  | { type: 'list'; content: ListContent }
  | { type: 'quote'; content: QuoteContent }
  | { type: 'image'; content: ImageContent }
  | { type: 'code'; content: CodeContent };

export type ContentBlock = ContentBlockData & {
  id: string;
};

export interface BlogPost {
  slug: string;
  title: string;
  excerpt: string;
  image?: string;
  date: string;
  readTime?: string;
  tags?: string[];
  content?: string;
  author: {
    name: string;
    avatar?: string;
  };
  comments?: number;
  contentBlocks?: ContentBlock[];
  contentType: 'markdown' | 'structured';
}
