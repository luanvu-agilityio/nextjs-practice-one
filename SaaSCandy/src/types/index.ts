export * from './user';
export * from './auth';
export * from './service';
export * from './plan';
export * from './pricing-plan';
export * from './faq';
