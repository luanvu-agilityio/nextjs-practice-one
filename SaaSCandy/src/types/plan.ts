export type Plan = {
  name: string;
  price: string | number;
  description: string;
  features: string[];
  popular?: boolean;
};

export type PricingPlanCardProps = {
  plan: Plan;
};
