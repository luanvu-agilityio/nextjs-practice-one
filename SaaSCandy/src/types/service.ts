export interface ServiceFeature {
  id: string;
  title: string;
  description: string;
}

export interface Service {
  slug: string;
  title: string;
  subtitle: string;
  description: string;
  whatItDoes: {
    title: string;
    description: string;
    image: string;
  };
  features: ServiceFeature[];
  icon: string;
}
