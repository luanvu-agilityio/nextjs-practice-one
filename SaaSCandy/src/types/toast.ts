export const TOAST_VARIANTS = {
  SUCCESS: 'success',
  ERROR: 'error',
  WARNING: 'warning',
  INFO: 'info',
};
export type ToastVariant = (typeof TOAST_VARIANTS)[keyof typeof TOAST_VARIANTS];

export interface ToastProps {
  id?: string | number;
  title: string;
  description?: string;
  variant?: ToastVariant;
  icon?: React.ReactNode;
  hasCloseIcon?: boolean;
  duration?: number;
}
