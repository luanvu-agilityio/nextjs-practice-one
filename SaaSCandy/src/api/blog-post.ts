import { BlogPost } from '@/types/blog-post';

/**
 * BlogHttpClient is a lightweight HTTP client for interacting with the blog API.
 * It supports GET requests and builds URLs from a base URL and path.
 */
export class BlogHttpClient {
  private readonly baseUrl: string;

  /**
   * Creates a new BlogHttpClient instance.
   * @param baseUrl - The base URL for the blog API.
   */
  constructor(baseUrl: string) {
    this.baseUrl = baseUrl.replace(/\/+$/, '');
  }

  /**
   * Builds a full URL from the base URL and path.
   * @param path - The endpoint path or full URL.
   * @returns The full URL as a string.
   */
  private buildUrl(path: string) {
    if (path.startsWith('http')) return path;
    return `${this.baseUrl}/${path.replace(/^\/+/, '')}`;
  }

  /**
   * Makes a GET request to the specified path.
   * @param path - The endpoint path or full URL.
   * @returns The parsed response data.
   * @throws {Error} If the response is not OK.
   */
  async get<T>(path: string): Promise<T> {
    const url = this.buildUrl(path);

    try {
      const response = await fetch(url, {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
      });

      if (!response.ok) {
        const errorText = await response.text();
        throw new Error(
          `HTTP error! status: ${response.status}, message: ${errorText}`
        );
      }

      return await response.json();
    } catch (error) {
      throw error;
    }
  }
}

// Create blog HTTP client using the separate blog API
const BLOG_API_BASE =
  process.env.NEXT_PUBLIC_API_BLOG_POSTS ||
  'https://68d374c3214be68f8c65df1d.mockapi.io/';

const blogHttp = new BlogHttpClient(BLOG_API_BASE);

/**
 * blogService provides methods to interact with blog post resources via the blog API.
 * Includes fetching all posts, fetching by slug or ID, pagination, filtering by tag, and searching.
 */
export const blogService = {
  /**
   * Fetches all blog posts.
   * @returns An array of BlogPost objects.
   * @throws {Error} If fetching fails or response format is invalid.
   */
  getAllPosts: async (): Promise<BlogPost[]> => {
    try {
      const response = await fetch(
        `${process.env.NEXT_PUBLIC_API_BLOG_POSTS}blog-posts`,
        {
          signal: AbortSignal.timeout(10000),
        }
      );

      if (!response.ok) {
        throw new Error(`Failed to fetch blog posts: ${response.status}`);
      }

      const posts = await response.json();

      if (!Array.isArray(posts)) {
        throw new Error('Invalid response format');
      }

      return posts;
    } catch (error) {
      console.error('Error fetching blog posts:', error);
      return [];
    }
  },

  /**
   * Fetches a blog post by its slug.
   * @param slug - The unique slug identifier for the blog post.
   * @returns The BlogPost object if found, otherwise null.
   * @throws {Error} If fetching fails or response format is invalid.
   */
  getPostBySlug: async (slug: string): Promise<BlogPost | null> => {
    try {
      const response = await fetch(
        `${process.env.NEXT_PUBLIC_API_BLOG_POSTS}blog-posts?slug=${slug}`,
        {
          signal: AbortSignal.timeout(10000),
        }
      );

      if (!response.ok) {
        throw new Error(`Failed to fetch blog post: ${response.status}`);
      }

      const posts = await response.json();

      if (!Array.isArray(posts)) {
        throw new Error('Invalid response format');
      }

      return posts.length > 0 ? posts[0] : null;
    } catch (error) {
      console.error(`Error fetching blog post with slug ${slug}:`, error);
      throw error;
    }
  },

  /**
   * Fetches a blog post by its ID.
   * @param id - The unique ID of the blog post.
   * @returns The BlogPost object if found, otherwise null.
   */
  getPostById: async (id: string): Promise<BlogPost | null> => {
    try {
      return await blogHttp.get<BlogPost>(`blog-posts/${id}`);
    } catch (error) {
      console.error(`Error fetching blog post with id ${id}:`, error);
      return null;
    }
  },

  /**
   * Fetches blog posts that match a specific tag.
   * @param tag - The tag to filter blog posts by.
   * @returns An array of BlogPost objects that contain the specified tag.
   */
  getPostsByTag: async (tag: string): Promise<BlogPost[]> => {
    try {
      const posts = await blogHttp.get<BlogPost[]>('blog-posts');
      return posts.filter(post =>
        post.tags?.some(postTag => postTag.toLowerCase() === tag.toLowerCase())
      );
    } catch (error) {
      console.error(`Error fetching blog posts with tag ${tag}:`, error);
      return [];
    }
  },

  /**
   * Searches blog posts by a query string.
   * @param query - The search term to match against title, excerpt, or tags.
   * @returns An array of BlogPost objects matching the query.
   */
  searchPosts: async (query: string): Promise<BlogPost[]> => {
    try {
      const posts = await blogHttp.get<BlogPost[]>('blog-posts');
      const searchTerm = query.toLowerCase();

      return posts.filter(
        post =>
          post.title.toLowerCase().includes(searchTerm) ||
          post.excerpt.toLowerCase().includes(searchTerm) ||
          post.tags?.some(tag => tag.toLowerCase().includes(searchTerm))
      );
    } catch (error) {
      console.error(`Error searching blog posts with query ${query}:`, error);
      return [];
    }
  },
};
