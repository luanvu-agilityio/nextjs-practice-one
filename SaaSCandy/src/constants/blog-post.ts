import { BlogPost } from '@/types/blog-post';

export const POSTS: BlogPost[] = [
  {
    slug: 'business-from-the-brink-of-ruin',
    title: 'Business from the brink of ruin by optimizing our website',
    excerpt:
      'Discover how we transformed our failing business through strategic website optimization and digital marketing.',
    image: '/images/blog-post/blog-post-1.png',
    date: '26 Dec 2025',
    readTime: '6 min',
    tags: ['seo', 'growth'],
    contentType: 'structured',
    contentBlocks: [
      {
        id: '1',
        type: 'heading',
        content: {
          level: 2,
          text: 'Main Heading & Points',
        },
      },
      {
        id: '2',
        type: 'paragraph',
        content: {
          text: 'Lorem ipsum is simply dummy text of the printing and typesetting industry...',
        },
      },
      {
        id: '3',
        type: 'list',
        content: {
          items: [
            'Vestibulum viverra hendrerit placerat consectetur tortor',
            'Etiam in velit ac augue pharetra',
            'Nullam dignissim blandit magna tellus, sit amet elit porta',
          ],
          ordered: false,
        },
      },
      {
        id: '4',
        type: 'quote',
        content: {
          text: 'We are a dedicated team of passionate product managers, developers, UX/UI designers.',
          author: 'Team SaaSCandy',
        },
      },
    ],
    author: {
      name: 'Alex Morgan',
      avatar: '/images/avatar.png',
    },
  },
  {
    slug: 'top-notch-work-processes',
    title:
      'The work is top-notch and I consistently outrank all my competitors',
    excerpt:
      'How a consistent process and focus on quality accelerated growth.',
    image: '/images/blog-post/blog-post-1.png',
    date: '24 Dec 2025',
    readTime: '4 min',
    tags: ['process', 'case-study'],
    contentType: 'structured',
    contentBlocks: [
      {
        id: '1',
        type: 'heading',
        content: {
          level: 2,
          text: 'Main Heading & Points',
        },
      },
      {
        id: '2',
        type: 'paragraph',
        content: {
          text: 'Lorem ipsum is simply dummy text of the printing and typesetting industry...',
        },
      },
      {
        id: '3',
        type: 'list',
        content: {
          items: [
            'Vestibulum viverra hendrerit placerat consectetur tortor',
            'Etiam in velit ac augue pharetra',
            'Nullam dignissim blandit magna tellus, sit amet elit porta',
          ],
          ordered: false,
        },
      },
      {
        id: '4',
        type: 'quote',
        content: {
          text: 'We are a dedicated team of passionate product managers, developers, UX/UI designers.',
          author: 'Team SaaSCandy',
        },
      },
    ],
    author: {
      name: 'Benjamin Lee',
      avatar: '/images/avatar.png',
    },
  },
  {
    slug: 'organic-search-marketing',
    title: 'Grow my business through organic search and marketing',
    excerpt: 'Tactical SEO and content playbook that actually worked.',
    image: '/images/blog-post/blog-post-1.png',
    date: '24 Dec 2025',
    readTime: '5 min',
    tags: ['seo', 'content'],
    contentType: 'structured',
    contentBlocks: [
      {
        id: '1',
        type: 'heading',
        content: {
          level: 2,
          text: 'Main Heading & Points',
        },
      },
      {
        id: '2',
        type: 'paragraph',
        content: {
          text: 'Lorem ipsum is simply dummy text of the printing and typesetting industry...',
        },
      },
      {
        id: '3',
        type: 'list',
        content: {
          items: [
            'Vestibulum viverra hendrerit placerat consectetur tortor',
            'Etiam in velit ac augue pharetra',
            'Nullam dignissim blandit magna tellus, sit amet elit porta',
          ],
          ordered: false,
        },
      },
      {
        id: '4',
        type: 'quote',
        content: {
          text: 'We are a dedicated team of passionate product managers, developers, UX/UI designers.',
          author: 'Team SaaSCandy',
        },
      },
    ],
    author: {
      name: 'Timothy Chen',
      avatar: '/images/avatar.png',
    },
  },
  {
    slug: 'tailored-brand-awareness',
    title: 'Tailored to your audience to enhance brand awareness',
    excerpt: 'Design and messaging tips to connect with your audience.',
    image: '/images/blog-post/blog-post-1.png',
    date: '24 Dec 2025',
    readTime: '3 min',
    tags: ['brand', 'design'],
    contentType: 'structured',
    contentBlocks: [
      {
        id: '1',
        type: 'heading',
        content: {
          level: 2,
          text: 'Main Heading & Points',
        },
      },
      {
        id: '2',
        type: 'paragraph',
        content: {
          text: 'Lorem ipsum is simply dummy text of the printing and typesetting industry...',
        },
      },
      {
        id: '3',
        type: 'list',
        content: {
          items: [
            'Vestibulum viverra hendrerit placerat consectetur tortor',
            'Etiam in velit ac augue pharetra',
            'Nullam dignissim blandit magna tellus, sit amet elit porta',
          ],
          ordered: false,
        },
      },
      {
        id: '4',
        type: 'quote',
        content: {
          text: 'We are a dedicated team of passionate product managers, developers, UX/UI designers.',
          author: 'Team SaaSCandy',
        },
      },
    ],
    author: {
      name: 'Trent Harris',
      avatar: '/images/avatar.png',
    },
  },
  {
    slug: 'strategic-business-insights',
    title: 'Uncover valuable insights for strategic business decisions',
    excerpt: 'Analytics and measurement practices for better decisions.',
    image: '/images/blog-post/blog-post-1.png',
    date: '24 Dec 2025',
    readTime: '6 min',
    tags: ['analytics'],
    contentType: 'structured',
    contentBlocks: [
      {
        id: '1',
        type: 'heading',
        content: {
          level: 2,
          text: 'Main Heading & Points',
        },
      },
      {
        id: '2',
        type: 'paragraph',
        content: {
          text: 'Lorem ipsum is simply dummy text of the printing and typesetting industry...',
        },
      },
      {
        id: '3',
        type: 'list',
        content: {
          items: [
            'Vestibulum viverra hendrerit placerat consectetur tortor',
            'Etiam in velit ac augue pharetra',
            'Nullam dignissim blandit magna tellus, sit amet elit porta',
          ],
          ordered: false,
        },
      },
      {
        id: '4',
        type: 'quote',
        content: {
          text: 'We are a dedicated team of passionate product managers, developers, UX/UI designers.',
          author: 'Team SaaSCandy',
        },
      },
    ],
    author: {
      name: 'Black Widow',
      avatar: '/images/avatar.png',
    },
  },
  {
    slug: 'drive-growth-achieve-objectives',
    title: 'Drive growth and achieve your business objectives effectively',
    excerpt: 'Roadmap to scale through product and marketing alignment.',
    image: '/images/blog-post/blog-post-1.png',
    date: '24 Dec 2025',
    readTime: '7 min',
    tags: ['growth', 'strategy'],
    contentType: 'structured',
    contentBlocks: [
      {
        id: '1',
        type: 'heading',
        content: {
          level: 2,
          text: 'Main Heading & Points',
        },
      },
      {
        id: '2',
        type: 'paragraph',
        content: {
          text: 'Lorem ipsum is simply dummy text of the printing and typesetting industry...',
        },
      },
      {
        id: '3',
        type: 'list',
        content: {
          items: [
            'Vestibulum viverra hendrerit placerat consectetur tortor',
            'Etiam in velit ac augue pharetra',
            'Nullam dignissim blandit magna tellus, sit amet elit porta',
          ],
          ordered: false,
        },
      },
      {
        id: '4',
        type: 'quote',
        content: {
          text: 'We are a dedicated team of passionate product managers, developers, UX/UI designers.',
          author: 'Team SaaSCandy',
        },
      },
    ],
    author: {
      name: 'Thomas Shelby',
      avatar: '/images/avatar.png',
    },
  },
  {
    slug: 'global-partner-network',
    title: 'Free access to our global partner network and marketplace',
    excerpt: 'How partnerships unlocked new channels and revenue.',
    image: '/images/blog-post/blog-post-1.png',
    date: '24 Dec 2025',
    readTime: '5 min',
    tags: ['partnerships'],
    contentType: 'structured',
    contentBlocks: [
      {
        id: '1',
        type: 'heading',
        content: {
          level: 2,
          text: 'Main Heading & Points',
        },
      },
      {
        id: '2',
        type: 'paragraph',
        content: {
          text: 'Lorem ipsum is simply dummy text of the printing and typesetting industry...',
        },
      },
      {
        id: '3',
        type: 'list',
        content: {
          items: [
            'Vestibulum viverra hendrerit placerat consectetur tortor',
            'Etiam in velit ac augue pharetra',
            'Nullam dignissim blandit magna tellus, sit amet elit porta',
          ],
          ordered: false,
        },
      },
      {
        id: '4',
        type: 'quote',
        content: {
          text: 'We are a dedicated team of passionate product managers, developers, UX/UI designers.',
          author: 'Team SaaSCandy',
        },
      },
    ],
    author: {
      name: 'George Martin',
      avatar: '/images/avatar.png',
    },
  },
  {
    slug: 'stages-of-work-priorities',
    title: 'Define each stage of work to see whatâ€™s important for you',
    excerpt: 'Prioritization framework for product and marketing teams.',
    image: '/images/blog-post/blog-post-1.png',
    date: '24 Dec 2025',
    readTime: '4 min',
    tags: ['product', 'process'],
    contentType: 'structured',
    contentBlocks: [
      {
        id: '1',
        type: 'heading',
        content: {
          level: 2,
          text: 'Main Heading & Points',
        },
      },
      {
        id: '2',
        type: 'paragraph',
        content: {
          text: 'Lorem ipsum is simply dummy text of the printing and typesetting industry...',
        },
      },
      {
        id: '3',
        type: 'list',
        content: {
          items: [
            'Vestibulum viverra hendrerit placerat consectetur tortor',
            'Etiam in velit ac augue pharetra',
            'Nullam dignissim blandit magna tellus, sit amet elit porta',
          ],
          ordered: false,
        },
      },
      {
        id: '4',
        type: 'quote',
        content: {
          text: 'We are a dedicated team of passionate product managers, developers, UX/UI designers.',
          author: 'Team SaaSCandy',
        },
      },
    ],
    author: {
      name: 'Luther King',
      avatar: '/images/avatar.png',
    },
  },
  {
    slug: 'plan-aligns-with-goals',
    title: 'We have a plan that aligns perfectly with your goals',
    excerpt: 'Planning and OKR examples that actually produce results.',
    image: '/images/blog-post/blog-post-1.png',
    date: '24 Dec 2025',
    readTime: '5 min',
    tags: ['planning', 'okr'],
    contentType: 'structured',
    contentBlocks: [
      {
        id: '1',
        type: 'heading',
        content: {
          level: 2,
          text: 'Main Heading & Points',
        },
      },
      {
        id: '2',
        type: 'paragraph',
        content: {
          text: 'Lorem ipsum is simply dummy text of the printing and typesetting industry...',
        },
      },
      {
        id: '3',
        type: 'list',
        content: {
          items: [
            'Vestibulum viverra hendrerit placerat consectetur tortor',
            'Etiam in velit ac augue pharetra',
            'Nullam dignissim blandit magna tellus, sit amet elit porta',
          ],
          ordered: false,
        },
      },
      {
        id: '4',
        type: 'quote',
        content: {
          text: 'We are a dedicated team of passionate product managers, developers, UX/UI designers.',
          author: 'Team SaaSCandy',
        },
      },
    ],
    author: {
      name: 'Lucas Brown',
      avatar: '/images/avatar.png',
    },
  },
];
