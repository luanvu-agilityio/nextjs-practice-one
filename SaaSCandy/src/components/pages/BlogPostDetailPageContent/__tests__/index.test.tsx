import { render } from '@testing-library/react';
import { BlogPostDetailPageContent } from '../index';
import { BlogPost } from '@/types/blog-post';

describe('BlogPostDetailPage Component', () => {
  const mockPost: BlogPost = {
    slug: 'test-post',
    title: 'Test Blog Post',
    excerpt: 'Test excerpt',
    content: '<p>Test content</p>',
    date: '2025-01-01',
    image: '/test.jpg',
    tags: ['javascript', 'react'],
    comments: 10,
    contentType: 'markdown',
    author: {
      name: 'John Doe',
      avatar: '/avatar.jpg',
    },
  };

  it('matches snapshot', () => {
    const { container } = render(<BlogPostDetailPageContent post={mockPost} />);
    expect(container).toMatchSnapshot();
  });

  it('renders post title and author', () => {
    const { getByText } = render(<BlogPostDetailPageContent post={mockPost} />);

    expect(getByText('Test Blog Post')).toBeInTheDocument();
    expect(getByText('John Doe')).toBeInTheDocument();
    expect(getByText('10 Comments')).toBeInTheDocument();
  });
});
