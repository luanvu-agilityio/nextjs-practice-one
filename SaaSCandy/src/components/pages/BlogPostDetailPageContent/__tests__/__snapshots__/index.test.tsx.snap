// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BlogPostDetailPage Component matches snapshot 1`] = `
<div>
  <section
    class="py-12 sm:py-16 bg-white"
  >
    <div
      class="max-w-[1296px] mx-auto px-4 sm:px-6 lg:px-0 "
    >
      <div
        class="flex items-center gap-2 sm:gap-4 text-xs sm:text-sm text-gray-background mb-4 sm:mb-7"
      >
        <span>
          2025-01-01
        </span>
        <span>
          10
           Comments
        </span>
      </div>
      <div
        class="flex flex-col lg:flex-row lg:items-center lg:justify-between gap-4 lg:gap-6 mb-4 sm:mb-6"
      >
        <h2
          aria-level="2"
          class="font-bold text-2xl sm:text-4xl lg:text-5xl"
          role="heading"
        >
          Test Blog Post
        </h2>
        <div
          class="flex items-center gap-3 sm:gap-6 lg:w-76.5 lg:justify-end"
        >
          <img
            alt="John Doe"
            class="rounded-full w-12 h-12 sm:w-16 sm:h-16 lg:w-20 lg:h-20"
            data-nimg="1"
            decoding="async"
            height="80"
            loading="lazy"
            src="/_next/image?url=%2Favatar.jpg&w=256&q=75"
            srcset="/_next/image?url=%2Favatar.jpg&w=96&q=75 1x, /_next/image?url=%2Favatar.jpg&w=256&q=75 2x"
            style="color: transparent;"
            width="80"
          />
          <div
            class="flex flex-col"
          >
            <span
              class=" sm:text-lg lg:text-xl font-bold"
            >
              John Doe
            </span>
            <span
              class="text-sm sm:text-xs lg:text-xl font-regular text-gray-background"
            >
              Author
            </span>
          </div>
        </div>
      </div>
      <div
        class="mb-6 sm:mb-8"
      >
        <img
          alt="Test Blog Post"
          class="w-full object-cover rounded-2xl sm:rounded-3xl"
          data-nimg="1"
          decoding="async"
          height="440"
          loading="lazy"
          src="/_next/image?url=%2Ftest.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Ftest.jpg&w=1920&q=75 1x, /_next/image?url=%2Ftest.jpg&w=3840&q=75 2x"
          style="color: transparent;"
          width="1296"
        />
      </div>
      <div
        class="flex flex-col lg:flex-row items-start gap-8 lg:gap-6"
      >
        <article
          class="flex-1 w-full lg:w-auto"
        >
          <div
            class="prose prose-sm sm:prose-base lg:prose-lg max-w-none mb-6 sm:mb-8"
          >
            <p>
              Test content
            </p>
          </div>
          <div
            class="border-t pt-6 sm:pt-8 mb-6 sm:mb-8"
          >
            <h3
              aria-level="3"
              class="text-xl sm:text-2xl font-bold text-gray-900 mb-3 sm:mb-4"
              role="heading"
            >
              Tags
            </h3>
            <div
              class=""
            >
              <h3
                aria-level="3"
                class="font-semibold text-gray-900 mb-2 sm:mb-3 text-lg sm:text-xl"
                role="heading"
              >
                Tags
              </h3>
              <ul
                class="space-y-1"
              >
                <li>
                  <span
                    class="text-xs sm:text-sm text-gray-600 capitalize"
                  >
                    - 
                    javascript
                  </span>
                </li>
                <li>
                  <span
                    class="text-xs sm:text-sm text-gray-600 capitalize"
                  >
                    - 
                    react
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="border-t pt-6 sm:pt-8 mb-6 sm:mb-8 lg:hidden"
          >
            <h3
              aria-level="3"
              class="text-xl sm:text-2xl font-bold text-gray-900 mb-3 sm:mb-4"
              role="heading"
            >
              Share
            </h3>
            <ul
              class="space-y-2"
            >
              <li>
                <a
                  class="text-sm sm:text-xs text-gray-600 hover:text-gray-900 transition-colors"
                  href="https://www.facebook.com/sharer/sharer.php?u=http%3A%2F%2Flocalhost%2F"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Facebook
                </a>
              </li>
              <li>
                <a
                  class="text-sm sm:text-xs text-gray-600 hover:text-gray-900 transition-colors"
                  href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%2F&text=Test%20Blog%20Post"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Twitter
                </a>
              </li>
              <li>
                <a
                  class="text-sm sm:text-xs text-gray-600 hover:text-gray-900 transition-colors"
                  href="https://www.linkedin.com/sharing/share-offsite/?url=http%3A%2F%2Flocalhost%2F"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  LinkedIn
                </a>
              </li>
            </ul>
          </div>
          <div
            class="border-t pt-6 sm:pt-8 lg:hidden"
          >
            <div
              class=""
            >
              <h3
                aria-level="3"
                class="font-semibold text-primary text-2xl sm:text-3xl lg:text-[35px] mb-4 sm:mb-6"
                role="heading"
              >
                Join our Newsletter
              </h3>
              <form
                class="space-y-3 sm:space-y-4"
              >
                <div
                  class="flex flex-col gap-2 w-full"
                >
                  <div
                    class="flex items-center h-11 min-w-0 transition-colors bg-transparent focus-within:border-white px-3 py-2 sm:px-4 sm:py-2.5 border border-primary rounded-lg text-sm sm:text-md focus:outline-none focus:ring-2 focus:ring-orange-500 focus:border-transparent w-full"
                  >
                    <input
                      aria-invalid="false"
                      class="w-full bg-transparent outline-none placeholder:text-placeholder disabled:pointer-events-none disabled:cursor-not-allowed"
                      id="input-«r0»"
                      placeholder="Email address Subscribe"
                      required=""
                      type="email"
                      value=""
                    />
                  </div>
                </div>
                <button
                  class="inline-flex items-center justify-center font-semibold text-md leading-none tracking-normal transition-colors duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed cursor-pointer bg-orange-background text-white hover:opacity-90 focus-visible:ring-primary px-9 py-5 rounded-lg w-full"
                  type="submit"
                >
                  Subscribe
                </button>
              </form>
            </div>
          </div>
        </article>
        <div
          class="hidden lg:block sticky top-4 w-104 py-12 px-11 space-y-12 shadow-form rounded-lg"
        >
          <div
            class="space-y-3 "
          >
            <h3
              aria-level="3"
              class="font-semibold text-primary text-2xl sm:text-3xl lg:text-[35px]"
              role="heading"
            >
              Share
            </h3>
            <div
              class="flex flex-wrap gap-2 sm:flex-col sm:gap-3"
            >
              <a
                class="flex gap-2 sm:gap-2.5 items-center py-1.5 px-3 sm:py-2 sm:px-4 rounded-lg text-white text-sm sm:text-md font-medium transition-colors bg-slate-blue hover:bg-teal-blue"
                href="https://www.facebook.com/sharer/sharer.php?u=http%3A%2F%2Flocalhost%2F"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-facebook w-3 h-3 sm:w-3.5 sm:h-3.5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18 2h-3a5 5 0 0 0-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 0 1 1-1h3z"
                  />
                </svg>
                Facebook
              </a>
              <a
                class="flex gap-2 sm:gap-2.5 items-center py-1.5 px-3 sm:py-2 sm:px-4 rounded-lg text-white text-sm sm:text-md font-medium transition-colors bg-sky-blue hover:bg-blue-foreground"
                href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%2F&text="
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-twitter w-3 h-3 sm:w-3.5 sm:h-3.5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M22 4s-.7 2.1-2 3.4c1.6 10-9.4 17.3-18 11.6 2.2.1 4.4-.6 6-2C3 15.5.5 9.6 3 5c2.2 2.6 5.6 4.1 9 4-.9-4.2 4-6.6 7-3.8 1.1 0 3-1.2 3-1.2z"
                  />
                </svg>
                Twitter
              </a>
              <a
                class="flex gap-2 sm:gap-2.5 items-center py-1.5 px-3 sm:py-2 sm:px-4 rounded-lg text-white text-sm sm:text-md font-medium transition-colors bg-teal-blue hover:bg-slate-blue"
                href="https://www.linkedin.com/sharing/share-offsite/?url=http%3A%2F%2Flocalhost%2F"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-linkedin w-3 h-3 sm:w-3.5 sm:h-3.5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
                  />
                  <rect
                    height="12"
                    width="4"
                    x="2"
                    y="9"
                  />
                  <circle
                    cx="4"
                    cy="4"
                    r="2"
                  />
                </svg>
                LinkedIn
              </a>
            </div>
          </div>
          <div
            class="w-full"
          >
            <h3
              aria-level="3"
              class="font-semibold text-primary text-2xl sm:text-3xl lg:text-[35px] mb-4 sm:mb-6"
              role="heading"
            >
              Join our Newsletter
            </h3>
            <form
              class="space-y-3 sm:space-y-4"
            >
              <div
                class="flex flex-col gap-2 w-full"
              >
                <div
                  class="flex items-center h-11 min-w-0 transition-colors bg-transparent focus-within:border-white px-3 py-2 sm:px-4 sm:py-2.5 border border-primary rounded-lg text-sm sm:text-md focus:outline-none focus:ring-2 focus:ring-orange-500 focus:border-transparent w-full"
                >
                  <input
                    aria-invalid="false"
                    class="w-full bg-transparent outline-none placeholder:text-placeholder disabled:pointer-events-none disabled:cursor-not-allowed"
                    id="input-«r1»"
                    placeholder="Email address Subscribe"
                    required=""
                    type="email"
                    value=""
                  />
                </div>
              </div>
              <button
                class="inline-flex items-center justify-center font-semibold text-md leading-none tracking-normal transition-colors duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed cursor-pointer bg-orange-background text-white hover:opacity-90 focus-visible:ring-primary px-9 py-5 rounded-lg w-full"
                type="submit"
              >
                Subscribe
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
